docs/source/conf.py

```diff
@@ -253,3 +257,22 @@ def process_readme():
 
 
 process_readme()
+
+# Auto-generate documentation pages
+current_dir = os.path.abspath(os.path.dirname(__file__))
+module = os.path.join(current_dir, "..", "..", "sentinelhub")
+
+APIDOC_EXCLUDE = [os.path.join(module, "commands.py")]
+APIDOC_OPTIONS = ["--module-first", "--separate", "--no-toc", "--templatedir", os.path.join(current_dir, "_templates")]
+
+
+def run_apidoc(_):
+    from sphinx.ext.apidoc import main
+
+    sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+
+    main(["-e", "-o", current_dir, module, *APIDOC_EXCLUDE, *APIDOC_OPTIONS])
```

Comment:
Is there a way that auto-generated files would be deleted at the end of this procedure? The problem is that when running docs build locally it leaves a lot of these files that we would probably accidentally be committing to git. 

Another option would be to put `sentinelhub.*.rst` files to `.gitignore`. If at any point we will want to commit any of them we'll commit with `-f`. Perhaps we could even put all of them into a subfolder of `source` folder so that they don't clutter the main docs folder?

Reply:
I would prefer to do it via .gitignore, otherwise we again need to handcraft a list of what to remove and what not to remove.
moving them around is difficult since the links between them are auto-generated

Reply:
Ok, I agree.

It might work if you put all of them and `sentinelhub.rst` into a subfolder. The only reference you would have to modify then is the reference from `index.rst` to `sentinelhub.rst`.

Reply:
Looks great. :bowing_man: